File: src/main/java/net/reflact/client/hud/overlay/ReflactOverlay.java
package net.reflact.client.hud.overlay;

import net.minecraft.client.gui.DrawContext;

public interface ReflactOverlay {
    void render(DrawContext context, float tickDelta);
    int getX();
    int getY();
    void setX(int x);
    void setY(int y);
    int getWidth();
    int getHeight();
    
    // Resize support
    default void setWidth(int width) {}
    default void setHeight(int height) {}
    
    boolean isEnabled();
    String getName();
}

File: src/main/kotlin/net/reflact/client/hud/overlay/ReflactOverlay.kt
package net.reflact.client.hud.overlay

import net.minecraft.client.gui.DrawContext

interface ReflactOverlay {
    fun render(context: DrawContext, tickDelta: Float)
    
    var x: Int
    var y: Int
    
    // Properties for resize/bounds
    var width: Int
        get() = 0 // Default impl
        set(value) {} // Default impl
        
    var height: Int
        get() = 0 // Default impl
        set(value) {} // Default impl

    val isEnabled: Boolean
    val name: String
}
